src/ray/object_manager/object_manager.cc

```diff
@@ -379,10 +387,34 @@ void ObjectManager::Push(const ObjectID &object_id, const ClientID &client_id) {
     return;
   }
 
+  // If we've pushed this same object to this same object manager recently, then
+  // don't do it again. Otherwise do it.
+  auto &recent_pushes = local_objects_[object_id].recent_pushes;
+  auto it = recent_pushes.find(client_id);
+  if (it == recent_pushes.end()) {
+    // We haven't pushed this specific object to this specific object manager
+    // yet (or if we have then the object must have been evicted and recreated
+    // locally).
+    recent_pushes[client_id] = current_time_ms();
+  } else {
+    // We've already pushed this object to this object manager, but if that
+    // happened long ago, push it again.
+    int64_t current_time = current_time_ms();
```

Comment:
Move this above the if block? Also, we should use `current_sys_time_ms` instead since it's not critical that this is a monotonic timestamp and `current_time_ms` is slower.

Reply:
Fixed